api/src/pcapi/scripts/move_offer/move_batch_offer.py

```diff
@@ -177,18 +181,24 @@ def _move_all_venue_offers(dry_run: bool, origin: int | None, destination: int |
         if invalidity_reason:
             invalid_venues.append((origin_venue_id, destination_venue_id, invalidity_reason))
         else:
-            offers_repository.lock_stocks_for_venue(origin_venue_id)
-            _move_individual_offers(origin_venue, destination_venue)
-            _move_collective_offers(origin_venue, destination_venue)
-            _move_collective_offer_template(origin_venue, destination_venue)
-            _move_collective_offer_playlist(origin_venue, destination_venue)
-            _move_price_category_label(origin_venue, destination_venue)
-            if not dry_run:
-                db.session.commit()
-                search.reindex_venue_ids([origin_venue_id])
-                logger.info("Transfert done for venue %d to venue %d", origin_venue_id, destination_venue_id)
-            else:
-                db.session.flush()
+            with atomic():
+                offers_repository.lock_stocks_for_venue(origin_venue_id)
+                _move_individual_offers(origin_venue, destination_venue)
+                _move_collective_offers(origin_venue, destination_venue)
+                _move_collective_offer_template(origin_venue, destination_venue)
+                _move_collective_offer_playlist(origin_venue, destination_venue)
+                _move_price_category_label(origin_venue, destination_venue)
+                if not dry_run:
+                    on_commit(
+                        partial(
+                            search.reindex_venue_ids,
+                            [origin_venue_id],
+                        )
+                    )
+                    logger.info("Transfert done for venue %d to venue %d", origin_venue_id, destination_venue_id)
```

Comment:
au passage tu peux corriger cette typo 

```suggestion
                    logger.info("Transfer done for venue %d to venue %d", origin_venue_id, destination_venue_id)
```